### ISSUE TITLE
Missing Test Case - 677. Map Sum Pairs

### ISSUE THREAD

**Issue by Sajal-1** [LeetCode-Feedback/LeetCode-Feedback#1565]
#### Your LeetCode username

Sajal3110

#### Category of the bug
- [ ] Question
- [ ] Solution
- [ ] Language

#### Description of the bug

Missing test case, which leads to acceptance for faulty code

#### Code you used for Submit/Run operation

```
class Node{
    int sum;
    Map<Character,Node> children;
    public Node(){
        sum=0;
        children=new HashMap<>();
    }
    
}
class MapSum {
    Node root;
    Map<String,Integer> map;
    /** Initialize your data structure here. */
    public MapSum() {
        root=new Node();
        map=new HashMap<>();
    }
    
    public void insert(String key, int val) {
        if(map.containsKey(key))
            val=val-map.get(key);
        else
            map.put(key,val);
        Node curr=root;
        for(int i=0; i<key.length();i++){
            char c=key.charAt(i);
            if(!curr.children.containsKey(c)){
                curr.children.put(c,new Node());
            }
            curr=curr.children.get(c);
            curr.sum+=val;
        }
    }
    
    public int sum(String key) {
        Node curr=root;
        for(int i=0; i<key.length();i++){
            char c=key.charAt(i);
            if(!curr.children.containsKey(c)){
                return 0;
            }
            curr=curr.children.get(c);
            
        }
        return curr.sum;
    }
}
```

#### Language used for code

Java

#### Expected behavior

The code gives wrong answer for this input, still passes all test cases provided.
["MapSum", "insert", "sum", "insert", "sum","insert","sum"]
[[], ["apple",3], ["ap"], ["apple",2], ["ap"], ["apple",4], ["ap"]]
Basically, the test cases do not check the case where multiple values for the same word (say, "apple") are provided more than 2 times.

**Comment by lc-kay** [comment]
@Sajal-1 When I ran your code to reproduce the issue, your code failed on test case 33. To contribute a missing test case, please provide an accepted code that it fails on. Then, I will reopen this issue. Thank you for your time!
